packages/grafana-ui/src/components/VizTooltip/VizTooltipFooter.tsx

```diff
@@ -79,21 +76,46 @@ const renderActions = makeRenderLinksOrActions<ActionModel>(
   (item, i) => <ActionButton key={i} action={item} variant="secondary" />
 );
 
-export const VizTooltipFooter = ({ dataLinks, actions = [], annotate }: VizTooltipFooterProps) => {
+export const VizTooltipFooter = ({ dataLinks, actions = [], annotate, onFilterClick }: VizTooltipFooterProps) => {
   const styles = useStyles2(getStyles);
   const hasOneClickLink = useMemo(() => dataLinks.some((link) => link.oneClick === true), [dataLinks]);
   const hasOneClickAction = useMemo(() => actions.some((action) => action.oneClick === true), [actions]);
 
+  const VIZ_TOOLTIP_FOOTER_BUTTONS = [
```

Comment:
this approach removes optimizations that existed in the previous code, let's tweak this.

Reply:
@leeoniya are you recommending to pull just the `renderDataLinks()` and `renderActions()` invocations out of the `.map()` in this PR, or to get rid of the `.map()` altogether?

Reply:
the way it is now is that it creates an array with objects, each has JSX nodes that can potentially be discarded based on shouldRender.

the previous code avoided allocating anything unless it was necessary to render it, and did not require a wrapping array and wrapper objects.

so yeah, i think we should use the previous approach and just append an adhoc filter rendering condition

Reply:
since this tooltip re-renders on potentially every mousemove event, this is codepath is hot and should not be wasteful.

Reply:
for barchart this isnt a big deal, bars are wide and few, and the tooltip core wont rerender the contents when the hovered value is same from one pixel to next.

but this is a shared component, and in timeseries panel it's significant